test/tests/functional/pbs_basil_support.py

```diff
@@ -362,5 +407,393 @@ def test_hbmemm_rsc(self):
+            self.logger.info('Expected ALPS Inventory Hook message found.')
+        else:
+            self.logger.info('Expected ALPS Inventory Hook message not found.')
+        self.assertTrue(rv)
+
+    def test_knl_batch_to_interactive(self):
+        """
+        Change the mode of any two KNL nodes to interactive. Then check if the
+        PBS_alps_inventory_check hook picks up on the change and nodes are
+        marked as stale. Restore changes to hook and mode of KNL nodes.
+        """
+        if self.du.platform == 'craysim':
+            self.skipTest(reason='xtprocadmin cmd is not on cray simulator')
+
+        # Find the list of KNL vnodes
+        klist = self.retklist()
+        if len(klist) == 0:
+            self.skipTest(reason='No KNL vnodes present')
+            return
+        else:
+            self.logger.info("KNL vnode list: %s" % (klist))
+
+        # Set pbshook frequency to 10 seconds
+        self.server.manager(MGR_CMD_SET, PBS_HOOK,
+                            {'enabled': 'true', 'freq': 10},
+                            id='PBS_alps_inventory_check', expect=True)
+
+        now = time.time()
+        # Change mode of two KNL nodes to interactive
+        if len(klist) >= 2:
+            k1 = klist[0]
+            k2 = klist[len(klist) - 1]
+            knl1 = int(re.search(r'\d+', k1).group())
+            knl2 = int(re.search(r'\d+', k2).group())
+
+        cmd = ['xtprocadmin', '-k', 'm', 'interactive', '-n', "%s" % (knl1)]
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        cmd = ['xtprocadmin', '-k', 'm', 'interactive', '-n', "%s" % (knl2)]
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        # Restart Mom with signal HUP
+        self.mom.signal('-HUP')
+
+        # Check that the nodes are now stale.
+        self.server.expect(VNODE, {'state': 'Stale'}, id=k1, interval=5)
+        self.server.expect(VNODE, {'state': 'Stale'}, id=k2)
+
+        # Change nodes back to batch mode and restart PBS
+        cmd = ['xtprocadmin', '-k', 'm', 'batch']
+        self.logger.info(cmd)
+        ret = self.server.du.run_cmd(self.server.hostname,
+                                     cmd, logerr=True)
+        self.assertEqual(ret['rc'], 0)
+
+        # Restart Mom with signal HUP
+        self.mom.signal('-HUP')
+
+        # Check that the nodes are now stale.
+        self.server.expect(VNODE, {'state': 'free'}, id=k1, interval=5)
+        self.server.expect(VNODE, {'state': 'free'}, id=k2)
+
+        # Restore hook freq to 300
+        self.server.manager(MGR_CMD_SET, PBS_HOOK,
+                            {'enabled': 'true', 'freq': 300},
+                            id='PBS_alps_inventory_check', expect=True)
```

Comment:
If your test fails or errors out, you won't get here to restore the hook.  Consider putting this in tearDown()